src/vs/workbench/contrib/debug/browser/debugToolBar.ts

```diff
@@ -230,12 +230,43 @@ export class DebugToolBar extends Themable implements IWorkbenchContribution {
 		if (y === undefined) {
 			y = this.storageService.getNumber(DEBUG_TOOLBAR_Y_KEY, StorageScope.PROFILE, 0);
 		}
-		const titleAreaHeight = 35;
-		if ((y < titleAreaHeight / 2) || (y > titleAreaHeight + titleAreaHeight / 2)) {
-			const moveToTop = y < titleAreaHeight;
-			this.setYCoordinate(moveToTop ? 0 : titleAreaHeight);
-			this.storageService.store(DEBUG_TOOLBAR_Y_KEY, moveToTop ? 0 : 2 * titleAreaHeight, StorageScope.PROFILE, StorageTarget.MACHINE);
+
+		this.setYCoordinate(y);
+	}
+
+	private setYCoordinate(y = this.yCoordinate): void {
+		const [yMin, yMax] = this.yRange;
+		y = Math.max(yMin, Math.min(y, yMax));
+		this.$el.style.top = `${y}px`;
+		this.yCoordinate = y;
+		this.storageService.store(DEBUG_TOOLBAR_Y_KEY, y, StorageScope.PROFILE, StorageTarget.MACHINE);
```

Comment:
This is writing to the StorageServer on every mousemove event. Should this be debounced?